introspection.go

```diff
@@ -86,24 +86,24 @@ func init() {
 		Fields: Fields{
 			"kind": &Field{
 				Type: NewNonNull(__TypeKind),
-				Resolve: func(p ResolveParams) interface{} {
+				Resolve: func(p ResolveParams) (interface{}, error) {
 					switch p.Source.(type) {
 					case *Scalar:
-						return TypeKindScalar
+						return TypeKindScalar, nil
 					case *Object:
-						return TypeKindObject
+						return TypeKindObject, nil
 					case *Interface:
-						return TypeKindInterface
+						return TypeKindInterface, nil
 					case *Union:
-						return TypeKindUnion
+						return TypeKindUnion, nil
 					case *Enum:
-						return TypeKindEnum
+						return TypeKindEnum, nil
 					case *InputObject:
-						return TypeKindInputObject
+						return TypeKindInputObject, nil
 					case *List:
-						return TypeKindList
+						return TypeKindList, nil
 					case *NonNull:
-						return TypeKindNonNull
+						return TypeKindNonNull, nil
 					}
 					panic(fmt.Sprintf("Unknown kind of type: %v", p.Source))
```

Comment:
I'd expect that when a resolver encounters an error in this new API they'd `return nil, fmt.Errorf(...)` instead of `panic(fmt.Sprintf(...))`. This would also move closer to a panic-less graphql-go.